inference-engine/src/CMakeLists.txt

```diff
@@ -45,3 +45,31 @@ add_custom_target(ie_libraries ALL
 if(NGRAPH_ONNX_IMPORT_ENABLE)
     add_dependencies(ie_libraries inference_engine_onnx_reader)
 endif()
+
+if(NOT BUILD_SHARED_LIBS)
+    target_link_libraries(inference_engine PRIVATE inference_engine_ir_v7_reader inference_engine_ir_reader)
+
+    if(NGRAPH_ONNX_IMPORT_ENABLE)
+        target_compile_definitions(inference_engine PRIVATE NGRAPH_ONNX_IMPORT_ENABLE)
+        target_link_libraries(inference_engine PRIVATE inference_engine_onnx_reader)
+    endif()
+
+    if(ENABLE_MKL_DNN)
+        ieTargetLinkWholeArchiveEx(inference_engine PUBLIC MKLDNNPlugin)
+    endif()
```

Comment:
BTW, can we implement the same scheme for reader and inference plugins? We have planned to implement the same logic with `plugins.xml` for reader plugins, but currently they are directly hardcoded into Inference Engine core.. 

@emmanuelattia-philips  do you think you have enough time to implement the same logic with plugins.xml for reader plugins? We can use the following scheme for plugins.xml:
```
<ie>                                                            
    <plugins>                                                   
        <plugin name="GNA" location="libGNAPlugin.so">          
        </plugin>                                               
        <plugin name="HETERO" location="libHeteroPlugin.so">    
        </plugin>                                               
        <plugin name="CPU" location="libMKLDNNPlugin.so">       
        </plugin>                                               
    </plugins>    
    <readers>                                                   
        <plugin name="IRv10" extensions="xml" location="libinference_engine_ir_reader.so">          
        </plugin>                                               
        <plugin name="IRv7" extensions="xml" location="libinference_engine_ir_v7_reader.so">    
        </plugin>                                               
        <plugin name="ONNX" extensions="onnx,prototxt" location="libinference_engine_onnx_reader.so">       
        </plugin>                                               
    </readers>                                                  
</ie>    
```

@ilyachur  are you OK with such scheme?

Reply:
Yes, I agree with your proposal

Reply:
I think it would be a good idea but it's a bit out of the scope of this PR, it would require more extensive refactoring between ie_core.cpp and ie_network_reader.cpp for the XML sharing.

Reply:
I agree that it's out of scope of this PR.. Just noted this because it allows to simplify logic how we load reader plugins for both static / dynamic case.